docs/how-to-build-with-ddn/with-stripe.mdx

```diff
@@ -0,0 +1,243 @@
+
+The build is stored as a set of JSON files in `engine/build`.
+
+### Step 7. Start your local services
+
+```sh title="Start your local Hasura DDN Engine and Stripe connector:"
+ddn run docker-start
+```
+
+Your terminal will be taken over by logs for the different services.
+
+### Step 8. Run your first query
+
+```sh title="In a new terminal tab, open your local console:"
+ddn console --local
+```
+
+```graphql title="In the GraphiQL explorer of the console, write this query:"
+query GET_ALL_PRODUCTS {
+  getProducts {
+    data {
+      id
+      name
+      description
+    }
+  }
+}
+```
+
+```json title="You'll get a response listing all of your products. For example:"
+{
+  "data": {
+    "getProducts": {
+      "data": [
+        {
+          "id": "3bef8a40-3c33-11ee-bb29-070df467ec94",
+          "name": "Tee",
+          "description": "A great t-shirt!"
+        },
+        {
+          "id": "e0a70b16-65b6-11ed-8788-8fa2504d64a3",
+          "name": "Sticker Sheet",
+          "description": "Who doesn't love stickers??"
+        },
+        {
+          "id": "a44eda7c-65b6-11ed-997b-53b5bdb7117e",
+          "name": "Hasuras in The Cloud Tee",
+          "description": "The cloud for your cloud."
+        },
+        {
+          "id": "8aa93f86-65b6-11ed-901c-f320d4e17bb2",
+          "name": "Dark Furry Logo Tee",
+          "description": "Furry for furries."
+        }
+      ]
+    }
+  }
+}
+```
+
+### Step 9. Add another command
+
+```sh title="Add the GetCheckoutSessions command:"
+ddn command add my_stripe "GetCheckoutSessions"
+```
+
+In `app/metadata` you'll find a newly-generated file: `GetCheckoutSessions.hml`. This will be used to expose customer
+checkout sessions and their related information via your supergraph API.
+
+### Step 10. Rebuild your project
+
+```sh title="Next, create a new build:"
+ddn supergraph build local
+```
+
+```sh title="Bring down the services by pressing CTRL+C and start them back up:"
+ddn run docker-start
+```
+
+### Step 11. Query your new build
+
+```graphql title="Head back to your console and query the GetCheckoutSessions endpoint:"
+query GET_CHECKOUT_SESSIONS {
+  getCheckoutSessions {
+    data {
+      id
+      amountTotal
+      customerDetails {
+        name
+        email
+      }
+    }
+  }
+}
+```
+
+```json title="Because of the schema of Stripe's API, you'll see results like this, which contain nested information, such as customer details:"
+{
+  "data": {
+    "getCheckoutSessions": {
+      "data": [
+        {
+          "id": "cs_test_<unique_identifier>",
+          "amountTotal": 1000,
+          "customerDetails": {
+            "name": "Rob",
+            "email": "robdemo@hasura.io"
+          }
+        },
+        {
+          "id": "cs_test_<unique_identifier>",
+          "amountTotal": 1000,
+          "customerDetails": {
+            "name": "Sandeep",
+            "email": "sandeepdemo@hasura.io"
+          }
+        }
+      ]
+    }
+  }
+}
+```
+
```

Comment:
Talking about relationships would be really useful and powerful here. Ie: relating a user or product in Stripe to one in Postgres or something. We talk about relationships in other how to build docs too.

Reply:
I agree and will add in some information about this; what I want to avoid here is adding a dependency on another data source + connector. I can make mention of it with an example, but we don't need to spin up PG as a part of this as well 👍

Reply:
No I only meant to draw attention to relationships here and how DDN will shine when this is added with a connector like Stripe. I didn't mean to have the user add PG or anything. Just understand how relationships enhance this.